apps/aecore/lib/aecore/contract/tx/contract_create_tx.ex

```diff
@@ -0,0 +1,337 @@
+defmodule Aecore.Contract.Tx.ContractCreateTx do
+  @moduledoc """
+  Contains the transaction structure for creating contracts
+  and functions associated with those transactions.
+  """
+
+  @behaviour Aecore.Tx.Transaction
+
+  alias __MODULE__
+  alias Aecore.Governance.GovernanceConstants
+  alias Aecore.Account.AccountStateTree
+  alias Aecore.Account.Account
+  alias Aecore.Contract.{Contract, Call, CallStateTree, ContractStateTree, Dispatch}
+  alias Aecore.Tx.Transaction
+  alias Aecore.Tx.DataTx
+  alias Aecore.Chain.Identifier
+
+  require Aecore.Contract.ContractConstants, as: Constants
+
+  @version 1
+
+  @typedoc "Reason of the error"
+  @type reason :: String.t()
+
+  @typedoc "Expected structure for the ContractCreate Transaction"
+  @type payload :: %{
+          code: binary(),
+          vm_version: byte(),
+          deposit: non_neg_integer(),
+          amount: non_neg_integer(),
+          gas: non_neg_integer(),
+          gas_price: non_neg_integer(),
+          call_data: binary()
+        }
+
+  @typedoc "Structure of the ContractCreate Transaction type"
+  @type t :: %ContractCreateTx{
+          code: binary(),
+          vm_version: byte(),
+          deposit: non_neg_integer(),
+          amount: non_neg_integer(),
+          gas: non_neg_integer(),
+          gas_price: non_neg_integer(),
+          call_data: binary()
+        }
+
+  @typedoc "Structure that holds specific transaction info in the chainstate."
+  @type tx_type_state() :: Chainstate.contracts()
+
+  @doc """
+  Definition of the ContractCreateTx structure
+
+  # Parameters
+  - code: the byte code of the contract
+  - vm_version: the VM/ABI to use
+  - deposit: held by the contract until it is deactivated (an even number, 0 is accepted)
+  - amount: to be added to the miner account
+  - gas: gas for the initial call
+  - gas_price: gas price for the call
+  - call_data: call data for the initial call (usually including a function name and args, interpreted by the contract)
+  """
+  defstruct [
+    :code,
+    :vm_version,
+    :deposit,
+    :amount,
+    :gas,
+    :gas_price,
+    :call_data
+  ]
+
+  use ExConstructor
```

Comment:
we wont be needing `ExConstructor` here if the init is implemented manually, right?